src/prefect/cli/deploy.py

```diff
@@ -451,7 +451,11 @@ async def _run_single_deploy(
 
     update_work_pool_image = False
 
-    if is_interactive() and not docker_build_step_exists:
+    build_step_set_to_null = (
+        True if "build" in deploy_config and deploy_config["build"] is None else False
+    )
```

Comment:
I think you can just assign the boolean here rather than having the if/else branching.

```suggestion
    build_step_set_to_null = "build" in deploy_config and deploy_config["build"] is None
```

Reply:
ah yeah, thanks!